src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/BrokeredServices/ServiceBrokerFactory.cs

```diff
@@ -48,30 +49,38 @@ public ServiceBrokerFactory(ExportProvider exportProvider)
     /// <summary>
     /// Returns a full-access service broker, but will return null if we haven't yet connected to the Dev Kit broker.
     /// </summary>
-    public IServiceBroker? TryGetFullAccessServiceBroker() => _container?.GetFullAccessServiceBroker();
+    public IServiceBroker? TryGetFullAccessServiceBroker() => _container.Task.IsCompletedSuccessfully ? _container.Task.Result?.GetFullAccessServiceBroker() : null;
+
+    /// <summary>
+    /// Returns a full-access service broker container, that consuming code can await on until we connected to the Dev Kit broker
+    /// </summary>
+    [Export(typeof(SVsBrokeredServiceContainer))]
```

Comment:
exposting SVsBrokeredServiceContainer as an async is not ideal and probably not recommended. Ideally we should define SVsFullAccessServiceBrokerAsync somewhere. But it is possible to define somewhere outside of Microsoft.ServiceHub.Framework that can be reference by both EA.D and Language Server?

Reply:
Never export to a contract name you don't control the namespace of. Otherwise you run the risk of this exported contract name colliding with one that the shell or other extensions do or may export in the future.
If you must export a shell object, do so with a custom contract name, like this:
```cs
[Export("Roslyn.BrokeredServiceContainer")]
public Task<IBrokeredServiceContainer> BrokeredServiceContainerAsync => ...
```

Reply:
Resolving the conversation after discussing this offline. Not issue here in LSP